pkg/activator/handler/handler.go

```diff
@@ -124,12 +126,27 @@ func (a *activationHandler) proxyRequest(revID types.NamespacedName, w http.Resp
 	}
 
 	var proxy *httputil.ReverseProxy
+	var proxyHandler http.Handler
 	if a.tls {
 		proxy = pkghttp.NewHeaderPruningReverseProxy(useSecurePort(target), hostOverride, activator.RevisionHeaders, true /* uss HTTPS */)
 	} else {
 		proxy = pkghttp.NewHeaderPruningReverseProxy(target, hostOverride, activator.RevisionHeaders, false /* use HTTPS */)
 	}
 
+	if enableHTTP1FullDuplex {
+		proxyWithFullDuplex := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			rc := http.NewResponseController(w)
+			_ = rc.EnableFullDuplex()
+			proxy.ServeHTTP(w, r)
+		})
+		tmpHandler := struct{ http.Handler }{}
```

Comment:
Is there a reason not to use the same pattern as in activator main (just wrapping the existing `proxy`)?